src/request.rs

```diff
@@ -64,19 +55,20 @@ impl<'a> Request<'a> {
 
         match self.request.operation() {
             // Filesystem initialization
-            ll::Operation::Init { arg } => {
-                let reply: ReplyRaw<fuse_init_out> = self.reply();
+            ll::Operation::Init(x) => {
+                let reply: ReplyRaw<abi::fuse_init_out> = self.reply();
                 // We don't support ABI versions before 7.6
-                if arg.major < 7 || (arg.major == 7 && arg.minor < 6) {
-                    error!("Unsupported FUSE ABI version {}.{}", arg.major, arg.minor);
+                let v = x.version();
```

Comment:
Could we use tuple deconstruction here, or a wrapper/new type around the version to provide accessors/ordering?

Reply:
Note: tuples do provide proper ordering, that's one of the advantages of using a tuple here.  See the next line: `v < (7, 6)` which was a big improvement on the previous `arg.major < 7 || (arg.major == 7 && arg.minor < 6)`.

I'll change the assignments below to use tuple deconstruction as you suggest.

Reply:
Hmm, this didn't work:

```rust
                (se.proto_major, se.proto_minor) = v;
```

Failed with:

```
error[E0070]: invalid left-hand side of assignment
  --> src/request.rs:68:50
   |
68 |                 (se.proto_major, se.proto_minor) = v;
   |                 -------------------------------- ^
   |                 |
   |                 cannot assign to this expression
   |
   = note: destructuring assignments are not currently supported
   = note: for more information, see https://github.com/rust-lang/rfcs/issues/372
```

I guess it only works with `let` :(.

Reply:
I added a newtype afterall